test/suites/test_datetime_arithmetic.py

```diff
@@ -0,0 +1,317 @@
+        'year': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3),
+            'arg_2': tarantool.Interval(year=1),
+            'res': tarantool.Datetime(year=2009, month=2, day=3),
+        },
+        'date': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3),
+            'arg_2': tarantool.Interval(year=1, month=2, day=3),
+            'res': tarantool.Datetime(year=2009, month=4, day=6),
+        },
+        'date_days_overflow': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3),
+            'arg_2': tarantool.Interval(year=1, month=2, day=30),
+            'res': tarantool.Datetime(year=2009, month=5, day=3),
+        },
+        'time': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3),
+            'arg_2': tarantool.Interval(hour=1, minute=2, sec=3),
+            'res': tarantool.Datetime(year=2008, month=2, day=3, hour=1, minute=2, sec=3),
+        },
+        'time_secs_overflow': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3),
+            'arg_2': tarantool.Interval(hour=1, minute=2, sec=13003),
+            'res': tarantool.Datetime(year=2008, month=2, day=3, hour=4, minute=38, sec=43),
+        },
+        'nsecs': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3, hour=3, minute=36, sec=43),
+            'arg_2': tarantool.Interval(nsec=10000023),
+            'res': tarantool.Datetime(year=2008, month=2, day=3, hour=3, minute=36, sec=43,
+                                      nsec=10000023),
+        },
+        'zero': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3, hour=3, minute=36, sec=43),
+            'arg_2': tarantool.Interval(),
+            'res': tarantool.Datetime(year=2008, month=2, day=3, hour=3, minute=36, sec=43),
+        },
+        'month_non_last_day_none_adjust': {
+            'arg_1': tarantool.Datetime(year=2009, month=1, day=30),
+            'arg_2': tarantool.Interval(month=13, adjust=tarantool.IntervalAdjust.NONE),
+            'res': tarantool.Datetime(year=2010, month=2, day=28),
+        },
+        'month_non_last_day_excess_adjust': {
+            'arg_1': tarantool.Datetime(year=2009, month=1, day=30),
+            'arg_2': tarantool.Interval(month=13, adjust=tarantool.IntervalAdjust.EXCESS),
+            'res': tarantool.Datetime(year=2010, month=3, day=2),
+        },
+        'month_non_last_day_last_adjust': {
+            'arg_1': tarantool.Datetime(year=2009, month=3, day=30),
+            'arg_2': tarantool.Interval(month=2, adjust=tarantool.IntervalAdjust.LAST),
+            'res': tarantool.Datetime(year=2009, month=5, day=30),
+        },
+        'month_overflow_last_day_last_adjust': {
+            'arg_1': tarantool.Datetime(year=2009, month=2, day=28),
+            'arg_2': tarantool.Interval(month=1, adjust=tarantool.IntervalAdjust.LAST),
+            'res': tarantool.Datetime(year=2009, month=3, day=31),
+        },
```

Comment:
Here you could add more cases:

1. month <= 0
2. NONE + src_days <= dst_days
3. EXCESS + src_days <= dst_days
4. LAST + src_days > dst_days

Reply:
> NONE + src_days <= dst_days
> EXCESS + src_days <= dst_days
> LAST + src_days > dst_days

Adjustment is not affect days addition since there is only one way to add days. Or am I misunderstood something?

Reply:
You know it. I know it. But the code does not know it. We can to test the behavior in the unit-test. It doesn't overcomplicate anything.

Reply:
month <= 0 now covered as positive month subtraction